packages/ens-controller/CHANGELOG.md

```diff
@@ -6,6 +6,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [4.0.0]
+### Changed
+- **BREAKING:** Bump to Node 16 ([#1262](https://github.com/MetaMask/core/pull/1262))
+- **BREAKING:** Add `@metamask/network-controller` as a dependency and peer dependency ([#1367](https://github.com/MetaMask/core/pull/1367), [#1362](https://github.com/MetaMask/core/pull/1362))
```

Comment:
I believe I've asked this question before, but why do we need this as a peer dependency if it's already a direct dependency?

Reply:
Well, it's for two reasons.

Firstly, when we're using the messaging system to communicate with a controller it _is_ a peer dependency. We're not communicating with whatever we have set as direct dependency - we're only getting the types from there. It's the controller instance setup in the client that we really talk to.

Second, it's so that the version we reference as a direct dependency matches what the client has setup. In practice the client will construct these controllers, so the expectations of this controller should match what the extension has. We won't be constructing them twice. Effectively they're used as singletons.

Reply:
Ah, right. I forgot that we were just getting the types from this package in this case. Okay makes sense why we'd need both.